ROADMAP.md

```diff
@@ -0,0 +1,118 @@
+
+How:
+
+1. Happy Eyeballs: implement some kind of “Happy-Eyeballs” style prioritization among all supported transports
+2. QUIC Blackhole detector: detection of blackholes, especially relevant to detect UDP (QUIC) blackholing
+3. RTT estimation: estimation of the expected RTT of a connection based on two nodes’ IP addresses, so that Happy Eyeballs Timeouts can be set dynamically
+
+## [C. 📊 Comprehensive Metrics](https://github.com/libp2p/go-libp2p/issues/1356)
+
+**Why**: For far too long, go-libp2p has been a black box. This has hurt us many times, by allowing trivial bugs to go undetected for a long time ([example](https://github.com/ipfs/kubo/pull/8750)). Having metrics will allow us to track the impact of performance improvements we make over time.
+
+**Goal**: Export a wider set of metrics across go-libp2p components and enable node operators to monitor their nodes in production. Optionally provide a sample Grafana dashboard similar to the resource manager dashboard.
+
+**How**: This will look similar to how we already expose resource manager metrics. Metrics can be added incrementally for libp2p’s components. First milestone is having metrics for the swarm.
+
+## [D. 📺 Universal Browser Connectivity](https://github.com/libp2p/go-libp2p/issues/1811)
+
+**Why**: A huge part of “the Web” is happening inside the browser. As a universal p2p networking stack, libp2p needs to be able to offer solutions for browser users.
+
+**Goal**: go-libp2p ships with up-to-date WebTransport and (libp2p-) WebRTC implementations, enabled by default. This allows connections between browsers and public nodes, browsers and non-public nodes as well as two browsers.
+
+1. [WebRTC](https://github.com/libp2p/go-libp2p/pull/1655): while browser to public node is getting close to finalized, there’ll be a push to make the other combinations work as well
+2. [WebTransport](https://github.com/libp2p/go-libp2p/issues/1717): as the protocol is still under development by IETF and W3C, our implementation needs to follow. To stay up to date, we will have to move as soon as Chrome ships support for a new draft version
+
+## [E. 🧪 Future-proof Testing](https://github.com/libp2p/go-libp2p/issues/1809)
+
+**Why**: Having lots of transports is great, and shaving off RTTs is awesome. We need to stay backwards-compatible with legacy go-libp2p implementations, and less advanced libp2p stacks.
+
+**Goal**: We have cross-version and cross-implementation Testground coverage that makes sure that we are able to establish a libp2p connection between two nodes, in the expected number of RTTs. This makes sure that the optimizations don’t break compatibility, actually have the performance impact we expect them to have, and serves as a regression test in the future.
+
+1. RTT test for Early Muxer Negotiation (A.1)
```

Comment:
I'm less convinced this warrants its own milestone.  This is basically part of the done criteria for early muxer integration negotiation.  All new functionality needs to be testing correctly as part of marking it complete.  I think it should be moved up.

I think we need:

Comprehensive interoperability testing across:
libp2p implementations (already happening)
implementation versions (already happening)
transports (I don't think we're exhaustive)
security/encryption (I don't think we're exhaustive)
muxers (I don't think we're exhaustive)

That is the key thing we want I believe so we can move forward with confidence that we're not breaking anything.